crates/ruff_formatter/src/format_element.rs

```diff
@@ -332,26 +332,26 @@ pub enum BestFittingMode {
 pub struct BestFittingVariants(Box<[FormatElement]>);
 
 impl BestFittingVariants {
-    /// Creates a new best fitting IR with the given variants. The method itself isn't unsafe
-    /// but it is to discourage people from using it because the printer will panic if
-    /// the slice doesn't contain at least the least and most expanded variants.
+    /// Creates a new best fitting IR with the given variants.
+    ///
+    /// Callers are required to ensure that the number of variants given
+    /// is at least 2.
     ///
     /// You're looking for a way to create a `BestFitting` object, use the `best_fitting![least_expanded, most_expanded]` macro.
     ///
-    /// ## Safety
-    /// The slice must contain at least two variants.
+    /// # Panics
+    ///
+    /// When the slice contains less than two variants.
     #[doc(hidden)]
-    #[allow(unsafe_code)]
-    pub unsafe fn from_vec_unchecked(variants: Vec<FormatElement>) -> Self {
-        debug_assert!(
+    pub fn from_vec_unchecked(variants: Vec<FormatElement>) -> Self {
+        assert!(
```

Comment:
Are you thinking about using a [`FormatError`](https://github.com/astral-sh/ruff/blob/f4d64720369e9de20bea76879a8f0693e1dc2b2c/crates/ruff_formatter/src/diagnostics.rs#L6-L27)? If so, which variant is appropriate here? I don't think I have enough context to write a good user facing error message here. (A panic looks appropriate to me here, since I imagine this is a bug in the code if the assert gets tripped. But I'm not completely certain.)

I suppose a behavior preserving change here would be to switch the `assert!` back to `debug_assert!` and document the panic conditions to `most_expanded` and `most_flat`.

Reply:
It would be a new variant (or two) in `InvalidDocumentError` but I'm also okay keeping the panics in `most_expanded` and `most_flat` and pointing it out in the documentation that Printing will fail with a panic if passed with less than 2 variations. 

The debug assertion mainly exists to make debugging easier, because you can identify the incorrect call from the call-stack which you can't when it panics during printing (and document inspection in the printer is a pain).